### PR TITLE
ReactiveAerospikeTemplate implementation (count-related methods)

### PR DESCRIPTION
**Pull request by igor-ermolenko**

### CODE REVIEW

**Comment by igor-ermolenko** [comment]
@Aloren @BrianNichols could you review this PR, please?

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java [hunk 1]
```diff
@@ -173,6 +175,33 @@ public String getSetName(Class<?> entityClass) {
                 });
     }
 
+    long countInternal(Query query, Class<?> type) {
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: why Internal? just `count` should be ok

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java [hunk 2]
```diff
@@ -173,6 +175,33 @@ public String getSetName(Class<?> entityClass) {
...
+        Assert.notNull(query, "Query must not be null!");
+        Assert.notNull(type, "Type must not be null!");
+
+        Qualifier qualifier = query.getCriteria().getCriteriaObject();
+        Stream<KeyRecord> results = findAllRecordsUsingQuery(type, null, qualifier);
+
+        return results.count();
+    }
+
+    long countInternal(Class<?> type, String setName) {
```
Comment (Aloren) [inline src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: this code is still blocking. not sure why we should add it into reactive template?

Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: I agree. Method removed.

File: src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java (hunk 1 above)
Comment (igor-ermolenko) [reply src/main/java/org/springframework/data/aerospike/core/BaseAerospikeTemplate.java]: OK. Reimplemented

**Comment by BrianNichols** [comment]
Merged into master branch.

**Comment by igor-ermolenko** [comment]
Thank you!

**Pull request merged**